### ISSUE TITLE
Configuration Clashing

### ISSUE THREAD

**Issue by wyattjoh** [mysticatea/npm-run-all#113]
We have a package.json:

```json
{
  "name": "talk",
  "scripts": {
    "a": "npm-run-all a:*",
    "a:a": "echo a",
    "a:b": "echo b",
    "lint": "echo lint"
  },
  "config": {
    "pre-git": {
      "commit-msg": [],
      "pre-commit": [
        "npm-run-all lint",
        "npm-run-all test"
      ],
      "pre-push": [
        "npm-run-all test"
      ],
      "post-commit": [],
      "post-merge": []
    }
  }
}
```

(truncated for simplicity)

And everytime I try to run `yarn a`, I get the following:

```
error Command "--talk:pre_git_pre_commit_0=npm-run-all lint" not found.
error Command "--talk:pre_git_pre_commit_0=npm-run-all lint" not found.
```

I suspect that the configuration generated by the https://github.com/bahmutov/pre-git package is interfereing with the configuration parsing that's done here:

https://github.com/mysticatea/npm-run-all/blob/7e83545dff5b3cd273823d2ebe5dbb17713887bd/lib/index.js#L101-L124

Possibly a way around this?

**Comment by mysticatea** [comment]
Thank you for the report.

Hmm, I could not reproduce it.

```bash
~\dev\sandbox> npm i -D npm-run-all
npm WARN talk@ No description
npm WARN talk@ No repository field.
npm WARN talk@ No license field.

+ npm-run-all@4.1.1
added 70 packages in 35.142s
~\dev\sandbox> npm run a

> talk@ a C:\Users\t-nagashima.MSS\dev\sandbox
> npm-run-all a:*


> talk@ a:a C:\Users\t-nagashima.MSS\dev\sandbox
> echo a

a

> talk@ a:b C:\Users\t-nagashima.MSS\dev\sandbox
> echo b

b

~\dev\sandbox> yarn a
yarn a v0.24.5
$ npm-run-all a:*
yarn run v0.24.5
$ echo a
a
Done in 0.08s.
yarn run v0.24.5
$ echo b
b
Done in 0.08s.
Done in 1.22s.
```

**Comment by mysticatea** [comment]
Oh, I can reproduce it with the latest `yarn`. I will investigate more.

**Comment by mysticatea** [comment]
`Yarn` seems to handle `--foo` as a task name in `yarn run --foo a`. This behavior is different to npm.

**Comment by wyattjoh** [comment]
Yeah, we're using yarn with our project, `1.2.1` to be exact.

**Comment by buz-zard** [comment]
I'm having the same issue with `yarn` version `1.2.1`:
```json
{
  "name": "npmrunall",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "config": {
    "msg_1": "message_c_1",
    "msg_2": "message_c_2"
  },
  "scripts": {
    "test0": "run-p test1 test2",
    "test1": "echo $npm_package_config_msg_1",
    "test2": "echo $npm_package_config_msg_2"
  },
  "devDependencies": {
    "npm-run-all": "^4.1.1"
  }
}
```
```
17:03 $ yarn test0
yarn run v1.2.1
$ run-p test1 test2
error Command "--npmrunall:msg_2=message_c_2" not found.
error Command "--npmrunall:msg_2=message_c_2" not found.
ERROR: "test1" exited with 1.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by mysticatea** [comment]
I have published `npm-run-all@4.1.2`. It should fix this problem.
